server/src/main/java/org/elasticsearch/action/support/replication/PostWriteRefresh.java

```diff
@@ -129,6 +144,7 @@ private void sendUnpromotableRequests(IndexShard indexShard, long generation, bo
             transportService.getLocalNode(),
             TransportUnpromotableShardRefreshAction.NAME,
             unpromotableReplicaRequest,
+            postWriteRefreshTimeout != null ? TransportRequestOptions.timeout(postWriteRefreshTimeout) : TransportRequestOptions.EMPTY,
```

Comment:
I'm not sure we need this if/else since TransportOptions supports null as a timeout.